src/routes/loginUtilities.js

```diff
@@ -7,7 +7,15 @@ module.exports.getAdminFromTokenOrRedirect = async (request, reply) => {
     }
 
     // Token must be valid
-    const admin = await getAdminByToken(request.cookies.auth);
-    if (!admin) reply.redirect('/login?auth=none');
-    else return admin;
+    let admin;
+    try {
+        admin = await getAdminByToken(request.cookies.auth);
+        if (!admin) reply.redirect('/login?auth=none');
+    } catch (error) {
+        if (error.message === 'jwt expired') {
```

Comment:
where does that come from?

Reply:
It's thrown in `src/sqlite/admin.js`. `getAdminByToken()` on line 123 calls `verifyToken()`, which then calls `jwt.verify()` on line 101. The jsonwebtoken module throws this error. There's also a custom error that gets thrown if `jwt.verify()` returns something falsy: "token invalid". I just checked the documentation, and `jwt.verify()` doesn't fail silently, so I'll remove that.

Is it better to bring the bcrypt and jwt work here into the routes? I'm used to the way Rails does routing, which keeps routing super separate from any logic.

Reply:
I did a bunch of commits, but didn't really touch the question above. I'm feeling like this stuff shouldn't be in the database code, and `routes/loginUtilities` is making more sense to me. I'll move it over.

Reply:
Okay, I'm pretty happy after moving that logic out of `src/sqlite/admin.js`. I've removed everything that touches the `Administrator_Tokens` table as well. I was being nervous when I added that. It ate up a lot of space. And if you think hard-coding a password is acceptable, I'm confident that this is just fine.

Can you please take another look at this before I merge?

Reply:
> Can you please take another look at this before I merge?

Sorry :sweat: you did good merging it

Reply:
No prob, bob! I really like the attitude you've had of "this is good enough, fix it when it breaks." I figure I got eyes on my code, I saw things from a new perspective, and if you have fixes/suggestions later, we can do them later. :D